src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,20 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (arg != NULL && arg[0] != '-') {
+      if (strcmp(arg, "map") == 0) {
+        code_env = CODE_ENVIRONMENT_DAEMON;
+        env_flag = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
+      }
+      break;
+    }
+  }
+
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, code_env, env_flag);
```

Comment:
I think it would be ok to me if we left your current version: it fixes [1] for the default case (when `admin socket` is not specified in ceph.conf). And if a user wanted to specify `admin socket` it would still need to make it unique (e.g. by adding $pid) to resolve conflict between `rbd-nbd map` processes. So with your current patch the behaviour is good enough, similar to what we have for `rbd` util. But if you add additional safeguard to reset `admin_socket` for not map commands, this will be ok for me to.

May be @dillaman has opinion

[1] http://tracker.ceph.com/issues/17951